### DISCUSSION TITLE
3.1 栈基础

### DISCUSSION THREAD

**Discussion by giscus** [itcharge/AlgoNote#140] [Comments]
# 3.1 栈基础

栈基础知识 1. 栈简介 栈（Stack）：一种线性表数据结构，是一种只允许在表的一端进行插入和删除操作的线性表。 我们把栈中允许插入和删除的一端称为 「栈顶（top）」；另一端则称为 「栈底（bottom）」。当表中没有任何数据元素时，称之为 「空栈」。栈有两种基本操作：「插入操作」 和 「删除操作」。 栈的插入操作又称为「入栈」...

[https://algo.itcharge.cn/03_stack_queue_hash_table/03_01_stack_basic/](https://algo.itcharge.cn/03_stack_queue_hash_table/03_01_stack_basic/)

**Comment by isooya** [comment]
勘误，链式栈的图片不太对吧？从左到右是a5到a1

**Comment by T1anq1** [comment]
后两张图片无法显示

**Comment by MortyCats** [comment]
这里建议中文直接写栈，不然容易跟堆搞混

**Reply by itcharge** [reply]
已修正歧义，感谢指正。

**Comment by Alexloco520** [comment]
"堆栈是算法和程序中最常用的辅助结构，其的应用十分广泛。堆栈基本应用于两个方面：

使用堆栈可以很方便的保存和取用信息，因此长被用作算法和程序中的辅助存储结构，临时保存信息，供后面操作中使用。"

-----------------------------------------------------------------------------------------------------------------------------

上面有个别字：“因此长被用作”。
这个“长”，应该是“常”。
